### ISSUE TITLE
False positive for CA1841 in Entity Framework queries

### ISSUE THREAD

**Issue by mikegoatly** [dotnet/roslyn-analyzers#5718]
### Analyzer

[CA1841](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1841) `Prefer Dictionary Contains methods`

### Analyzer source

[Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

6.0.0 (Latest)

### Describe the bug

Entity Framework queries can make use of lists when building queries, e.g.:

``` csharp
var dictionary = new Dictionary<int, SomeOtherType>() { ... };

var data = await this.dataContext.Products.Where(p => dictionary.Keys.Contains(p.Id)).ToListAsync();
```

However it can't make use of ContainsKey because it doesn't translate well to SQL.

I understand that this is probably an edge case, but it would be nice if the rule didn't fire in this case.

### Steps To Reproduce

Use `Dictionary.Keys` in an Entity Framework query

### Expected behavior

Ideally this wouldn't trigger the code analysis rule.

### Actual behavior

The code analysis rule is triggered, meaning in codebases that have warnings as errors enabled the code has to be re-written, or the rule excluded.

**Comment by Evangelink** [comment]
There is a helper method to check whether the target is `IEnumerable<T>` or `IQueryable<T>` so I think we could be on the safe side by disabling the rule for the latter.

@mavasani WDYT?
